src/test/java/uk/gov/pay/directdebit/payments/resources/PaymentResourceIT.java

```diff
@@ -311,6 +311,35 @@ public void shouldRetrieveAPayment_fromPublicApiEndpoint() {
                 .body(JSON_STATE_DETAILS_KEY, is("example_details"));
     }
     
+    @Test
+    public void shouldTriggerAnInvalidMandateStateException() throws Exception {
+        String accountExternalId = testGatewayAccount.getExternalId();
+        PayerFixture payerFixture = aPayerFixture();
+        MandateFixture mandateFixture = aMandateFixture()
+                .withGatewayAccountFixture(testGatewayAccount)
+                .withPayerFixture(payerFixture)
+                .withState(MandateState.CANCELLED)
+                .insert(testContext.getJdbi());
+
+        String postBody = new ObjectMapper().writeValueAsString(ImmutableMap.builder()
+                .put(JSON_AMOUNT_KEY, AMOUNT)
+                .put(JSON_REFERENCE_KEY, "Test description")
+                .put(JSON_DESCRIPTION_KEY, "Test description")
+                .put(JSON_GATEWAY_ACC_KEY, accountExternalId)
+                .put(JSON_MANDATE_ID_KEY, mandateFixture.getExternalId().toString())
+                .build());
+
+        String requestPath = "/v1/api/accounts/{accountId}/charges/collect"
+                .replace("{accountId}", accountExternalId);
+
+        ValidatableResponse response = givenSetup()
+                .body(postBody)
+                .post(requestPath)
+                .then()
+                .statusCode(500);
```

Comment:
should probably check the error_identifier here.

Reply:
Done